packages/server/src/subscriptions/websockets.ts

```diff
@@ -51,7 +61,35 @@ async function setupSubscriptionHandler(): Promise<void> {
     for (const [resource, subscriptionId, options] of subEventArgsArr) {
       const bundle = createSubEventNotification(resource, subscriptionId, options);
       for (const socket of subToWsLookup.get(subscriptionId) ?? []) {
-        socket.send(JSON.stringify(bundle), { binary: false });
+        // Get the repo for this socket in the context of the subscription
+        const subMetadataMap = wsToSubLookup.get(socket);
+        if (!subMetadataMap) {
+          // We should really never hit this log, this is a true error
+          globalLogger.error('[WS] Unable to find sub metadata map for WebSocket');
+          continue;
+        }
+        const subMetadata = subMetadataMap.get(subscriptionId);
+        if (!subMetadata) {
+          // We should really never hit this log, this is a true error
+          globalLogger.error('[WS] Unable to find sub metadata entry for WebSocket', { subscriptionId });
```

Comment:
Same here?

Reply:
Same as above I think, I think this one is maybe even less likely